language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FunctionTypeDescriptorNodeContext.java

```diff
@@ -52,10 +55,23 @@ public FunctionTypeDescriptorNodeContext() {
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, FunctionTypeDescriptorNode node) {
         List<LSCompletionItem> completionItems = new ArrayList<>();
         NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
-
-        if (this.onSuggestionsAfterQualifiers(context, node)) {
-            // Currently we consider the isolated qualifier only
-            completionItems.add(new SnippetCompletionItem(context, Snippet.KW_FUNCTION.get()));
+        if (onSuggestionsAfterQualifiers(context, node)) {
+            /*
+             * Covers the following
+             * type T <qualifier(s)> <cursor>
+             * Currently the qualifier can be isolated/transactional.
+             */
+            NodeList<Token> qualifiers = node.qualifierList();
+            List<Token> otherQuals = node.leadingInvalidTokens();
+            Token lastQualifier;
+            if (qualifiers.size() > 0) {
+                lastQualifier = qualifiers.get(qualifiers.size() - 1);
+            } else {
+                lastQualifier = otherQuals.get(otherQuals.size() - 1);
+            }
+            Set<SyntaxKind> qualKinds = qualifiers.stream().map(Token::kind).collect(Collectors.toSet());
+            qualKinds.addAll(otherQuals.stream().map(Token::kind).collect(Collectors.toSet()));
```

Comment:
Are we going to handle a generic type definition use-case here? If so why do we route the completion provider to a more generic one (`FunctionTypeDescriptorNodeContext`) here? We have to fix that routing also along with this fix.

Reply:
Not the generic type I have put `T` for type name. Will refactor this

Reply:
![desktop](https://user-images.githubusercontent.com/35211477/131671086-38030ea8-b104-429c-a1b1-7d66b74ea2d2.png)

Currently the resolved syntax tree for the respective scenario is as above. The node at cursor becomes the function type descriptor in this case. I suppose we are not able to tackle the problem in the routing approach.

Reply:
The comment I have put may have mislead you. Please take a look at the new comment.

Reply:
>Are we going to handle a generic type definition use-case here? If so why do we route the completion provider to a more generic one (FunctionTypeDescriptorNodeContext) here? We have to fix that routing also along with this fix.

Added Pre validation logic to fix this